src/_data/ghost.js

```diff
@@ -30,13 +36,47 @@ module.exports = async () => {
     )
   )
     .then(arr => {
-      console.log('Finished processing all posts');
+      console.log('Finished processing all Ghost posts');
       return arr.flat();
     })
     .catch(err => console.error(err));
+  const hashnodePosts = await Promise.all(
+    chunk(allHashnodePosts, batchSize).map((batch, i, arr) =>
+      piscinaHashnode.run({
+        batch,
+        currBatchNo: Number(i) + 1,
+        totalBatches: arr.length
+      })
+    )
+  )
+    .then(arr => {
+      console.log('Finished processing all Hashnode posts');
+      return arr.flat();
+    })
+    .catch(err => console.error(err));
+
+  const ghostPostSlugsSet = new Set(ghostPosts.map(post => post.slug));
+  const hashnodePostSlugsSet = new Set(hashnodePosts.map(post => post.slug));
+  const commonPostSlugs = [...ghostPostSlugsSet].filter(slug =>
+    hashnodePostSlugsSet.has(slug)
+  );
+
+  const posts = [...ghostPosts, ...hashnodePosts]
+    .filter(post => {
+      if (commonPostSlugs.includes(post.slug)) {
+        errorLogger({ type: 'duplicate', name: post.title });
+        return false;
+      }
+      return true;
+    })
+    .sort((a, b) => {
+      return new Date(b.published_at) - new Date(a.published_at);
+    });
```

Comment:
This was the behaviour we wanted from what I could gather from the dev meeting 2 weeks back ie. not build articles if slug matches from both source and report it.

cc: @raisedadead

Reply:
Yeah - it will be a hard problem but during the move the authors (or at least the publication leads will need to be on the watch and "fix" slugs). We can do a automated "notification" when this happens.

Reply:
To actively block a build and notify the editorial team to fix it ASAP. I want to reiterate that the build will fail even in our current Ghost setup. This has always been the case.

Slug clashes are presumably rare instances because the headlines are literally handed off by the editorial team. If a clash happens, it is a deeper problem that needs fixing.

Lastly, I want to emphasize that this check is only to be used during the transition phase, not before, not after.

Reply:
Yes.